src/c-writer.cc

```diff
@@ -1085,40 +1085,18 @@ void CWriter::WriteTags() {
   for (const Tag* tag : module_->tags) {
     bool is_import = tag_index < module_->num_tag_imports;
     if (!is_import) {
-      Write("static u32 ",
+      // Tags are identified and compared solely by their (unique) address.
+      // The data stored in this variable is never read.
+      if (tag_index == module_->num_tag_imports) {
+        Write(Newline());
+        Write("typedef bool wasm_tag_placeholder_t;", Newline());
```

Comment:
Maybe `char` here?   IIRC bool could be bigger than a char?  Also, can we use `bool` in this C89 output?

Reply:
Sure, char it is! (FWIW I think wasm2c's output has a pretty enmeshed dependency on C99 at present. CALL_INDIRECT is a variadic macro, we use the "inline" keyword, we have declarations after statements within the function bodies for load and store, we use "ll" integer constants, we use for loops with a variable declaration, etc. I don't see a reason we couldn't make it C89-clean but it would be some work.)

Reply:
Perhaps I meant C99.. I can't remember what goal we had exactly.   I guess one the problem is that MSVC never really implemented C99 IIRC.   Is `bool` and `stdbool.h` in C99?

Reply:
_Bool/bool/stdbool.h were added in C99, yeah. And I think you're right about MSVC. :-/

Reply:
Fwiw, I have always found the MSVC implemented enough of C99 to be useful (experience from other transpilers). I think MSVC recently fixed some of the c99 pre-processor handling as well, so I suspect C99 in general won't be a problem on recent MSVC compilers at least.